salt/_renderers/metalk8s_kubernetes.py

```diff
@@ -1,327 +1,108 @@
-    return {
-        'metalk8s_kubernetes.namespace_present': [
-            {'name': obj['metadata']['name']},
-            {'body': obj},
-            {'kubeconfig': kubeconfig},
-            {'context': context},
-        ],
-    }
-
-
-@handle('v1', 'Secret')
-def _handle_v1_secret(obj, kubeconfig, context):
-    # FIXME #1183: this is a compatibility band-aid for the vendored salt module
-    if 'data' in obj:
-        data = {
-            key: base64.b64decode(value)
-            for key, value in obj['data'].items()
-        }
-    elif 'stringData' in obj:
-        data = obj['stringData']
-    else:
-        data = None
-    return {
-        'metalk8s_kubernetes.secret_present': [
-            {'name': obj['metadata']['name']},
-            {'namespace': obj['metadata']['namespace']},
-            {'data': data},
-            {'kubeconfig': kubeconfig},
-            {'context': context},
-        ],
-    }
-
-
-@handle('monitoring.coreos.com/v1', 'Alertmanager')
-@handle('monitoring.coreos.com/v1', 'Prometheus')
-@handle('monitoring.coreos.com/v1', 'PrometheusRule')
-@handle('monitoring.coreos.com/v1', 'ServiceMonitor')
-def _handle_monitoring_coreos_com_v1_customresource(obj, kubeconfig, context):
-    return {
-        'metalk8s_kubernetes.customresource_present': [
-            {'name': obj['metadata']['name']},
-            {'body': obj},
-            {'kubeconfig': kubeconfig},
-            {'context': context},
-            {'namespace': obj['metadata']['namespace']},
-        ],
-    }
-
-
-@handle('apiregistration.k8s.io/v1', 'APIService')
-@handle('apiregistration.k8s.io/v1beta1', 'APIService')
-def _handle_apiregistration_v1_apiservice(obj, kubeconfig, context):
-    return {
-        'metalk8s_kubernetes.apiservice_present': [
-            {'name': obj['metadata']['name']},
-            {'metadata': obj['metadata']},
-            {'spec': obj['spec']},
-            {'kubeconfig': kubeconfig},
-            {'context': context},
-        ]
-    }
-
-
-@handle('policy/v1beta1', 'PodSecurityPolicy')
-@handle('extensions/v1beta1', 'PodSecurityPolicy')
-def _handle_extensions_v1beta1_podsecuritypolicy(obj, kubeconfig, context):
-    return {
-        'metalk8s_kubernetes.podsecuritypolicy_present': [
-            {'name': obj['metadata']['name']},
-            {'metadata': obj['metadata']},
-            {'spec': obj['spec']},
-            {'kubeconfig': kubeconfig},
-            {'context': context},
-        ]
-    }
-
-
-del handle
-
-
-def _step(obj, kubeconfig=None, context=None):
-    '''
-    Handle a single Kubernetes object, rendering it into a state 'step'
-    '''
-    name = _step_name(obj)
-    api_version = obj['apiVersion']
-    kind = obj['kind']
+def _step(manifest, kubeconfig=None, context=None, absent=False):
+    """Render a single Kubernetes object into a state 'step'."""
+    try:
+        obj = __utils__['metalk8s_kubernetes.convert_manifest_into_object'](
```

Comment:
This should be called `convert_manifest_to_object` (or even `manifest_to_object`).